packages/component/src/ActivityFeedback/ActivityFeedback.tsx

```diff
@@ -1,27 +1,69 @@
-import { WebChatActivity } from 'botframework-webchat-core';
-import React, { memo } from 'react';
+import classNames from 'classnames';
+import React, { memo, useCallback, type FormEventHandler, type KeyboardEventHandler } from 'react';
+import { Extract, wrapWith } from 'react-wrap-with';
+import { useRefFrom } from 'use-ref-from';
+
+import useStyleSet from '../hooks/useStyleSet';
 import FeedbackLoopWithMessage from './private/FeedbackLoopWithMessage';
 import FeedbackLoopWithoutMessage from './private/FeedbackLoopWithoutMessage';
 import ActivityFeedbackComposer from './providers/ActivityFeedbackComposer';
+import useFeedbackText from './providers/useFeedbackText';
+import useFocusAction from './providers/useFocusAction';
+import useSelectedAction from './providers/useSelectedAction';
 import useShouldShowFeedbackForm from './providers/useShouldShowFeedbackForm';
+import useSubmitCallback from './providers/useSubmitCallback';
+
+function InternalActivityFeedback() {
+  const [{ feedbackForm }] = useStyleSet();
+  const [feedbackText, setFeedbackText] = useFeedbackText();
+  const [selectedAction, setSelectedAction] = useSelectedAction();
+  const [shouldShowFeedbackForm] = useShouldShowFeedbackForm();
+  const focusAction = useFocusAction();
+  const submit = useSubmitCallback();
+
+  const feedbackTextRef = useRefFrom(feedbackText);
+  const selectedActionRef = useRefFrom(selectedAction);
 
-const InternalActivityFeedback = memo(() =>
-  useShouldShowFeedbackForm()[0] ? <FeedbackLoopWithMessage /> : <FeedbackLoopWithoutMessage />
-);
+  const handleReset = useCallback<FormEventHandler<HTMLFormElement>>(() => {
+    focusAction(selectedActionRef.current);
 
-InternalActivityFeedback.displayName = 'InternalActivityFeedback';
+    setFeedbackText(undefined);
+    setSelectedAction(undefined);
+  }, [focusAction, selectedActionRef, setFeedbackText, setSelectedAction]);
 
-type ActivityFeedbackProps = Readonly<{
-  activity: WebChatActivity;
-}>;
+  const handleSubmit = useCallback<FormEventHandler<HTMLFormElement>>(
+    event => {
+      event.preventDefault();
+
+      submit(selectedActionRef.current, feedbackTextRef.current);
+    },
+    [feedbackTextRef, selectedActionRef, submit]
+  );
+
+  const handleKeyDown = useCallback<KeyboardEventHandler<HTMLFormElement>>(
+    event => {
+      if (event.key === 'Escape' && selectedActionRef.current) {
+        event.stopPropagation();
+        event.currentTarget.reset();
+      }
+    },
+    [selectedActionRef]
+  );
 
-function ActivityFeedback({ activity }: ActivityFeedbackProps) {
   return (
-    <ActivityFeedbackComposer activity={activity}>
-      <InternalActivityFeedback />
-    </ActivityFeedbackComposer>
+    <form
+      className={classNames('webchat__feedback-form-real', feedbackForm + '')}
```

Comment:
`feedback-form-real` :grin:

Reply:
I was renaming `feedback-form` to `feedback-loop` and this `feedback-form-real` is `feedback-form`. Didn't finish my work yet. 🤣